core/src/main/kotlin/com/alecstrong/sql/psi/core/SqlFileBase.kt

```diff
@@ -167,18 +165,18 @@ abstract class SqlFileBase(
   }
 
   private inline fun iteratePreviousStatements(block: (SqlStmt) -> Unit) {
-    val files = SparseArrayCompat<SqlFileBase>()
+    val files = mutableMapOf<Int, SqlFileBase>()
     val topFiles = LinkedHashSet<SqlFileBase>()
     iterateSqlFiles { file ->
       if (file == this) return@iterateSqlFiles true
       else if (order != null && file.order == null) return@iterateSqlFiles true
       else if (order == null && file.order == null) topFiles.add(file)
-      else if (order == null || (file.order != null && file.order!! < order!!)) files.put(file.order!!, file)
+      else if (order == null || (file.order != null && file.order!! < order!!)) files[file.order!!] = file
 
       return@iterateSqlFiles true
     }
 
-    files.forEach { _, file ->
+    files.forEach { (_, file) ->
```

Comment:
will this already iterate them in order? (ie the lowest key value first). If not I think this just needs a `files.sorted.forEach`

Reply:
Ah yeah, it's been a minute since I've used the SparseArray from Android. Fixed in https://github.com/AlecStrong/sql-psi/pull/148/commits/5a29d3dae849c186c54ff788b415b28dbabf2e84.